README.md

```diff
@@ -231,6 +232,77 @@ Variables may also have _default_ values. We can define as such using the follow
 FOO=${FOO:-'default'}
 ```
 
+# Shell expansions
+
+_Expansions_ are performed on the command line after it has been split into _tokens_. In other words, these expansions are mechanism to calculate arithmetical operations, to save results of command's executions and so on.
+
+If you are interested, you can read [more about shell expansions](https://www.gnu.org/software/bash/manual/bash.html#Shell-Expansions).
+
+## Brace expansion
+
+Brace expansion allow us to generate arbitrary strings. It's similar to _filename expansion_. For example:
+
+```bash
+echo beg{i,a,u}n # begin began begun
+```
+
+Also brace expansions may be used for creating ranges, which are iterated over in loops.
+
+```bash
+echo {0..5} # 0 1 2 3 4 5
+echo {00..8..2} # 00 02 04 06 08
+```
+
+## Command substitution
+
+Command substitution allow us to evaluate a command and substitute its value into another command or variable assignment. Command substitution is performed when a command is enclosed by `` `​` `` or `$()`.  For example, we can use it as follows:
```

Comment:
use triple backquotes:
```md
``` `` ```
```

It will work well. We had a much pain with this.